python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -379,6 +385,73 @@ def do_evaluate(
                 ),
                 dtype=self.dtype,
             )
+        elif self.name is StringFunction.Name.SplitN:
+            n = self.options[0]
+            child, expr = self.children
+            column = child.evaluate(df, context=context)
+            if n == 1:
+                plc_column = plc.Column(
+                    self.dtype.plc,
+                    column.obj.size(),
+                    None,
+                    None,
+                    0,
+                    column.obj.offset(),
+                    [column.obj],
+                )
+            else:
+                assert isinstance(expr, Literal)
+                by = plc.Scalar.from_py(expr.value, expr.dtype.plc)
+                plc_table = plc.strings.split.split.split(
+                    column.obj,
+                    by,
+                    n - 1,
+                )
+                children = plc_table.columns()
+                ref_column = children[0]
+                if (remainder := n - len(children)) > 0:
+                    children.extend(
+                        plc.Column.all_null_like(ref_column, ref_column.size())
+                        for _ in range(remainder)
+                    )
+                plc_column = plc.Column(
+                    self.dtype.plc,
+                    ref_column.size(),
+                    None,
+                    None,
+                    0,
+                    ref_column.offset(),
+                    children,
+                )
+            return Column(plc_column, dtype=self.dtype)
+        elif self.name is StringFunction.Name.SplitExact:
```

Comment:
Is it worth trying to combine this case with the `n>1` case above? The two should be almost the same, right? `splitn` just has the last column contain all the leftovers (if there are any) instead of the final split result IIUC.